imap-codec/fuzz/fuzz_targets/fragmentizer.rs

```diff
@@ -0,0 +1,52 @@
+#![no_main]
+
+use imap_codec::fragmentizer::{Fragmentizer, MaxMessageSize};
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|input: (MaxMessageSize, Vec<&[u8]>)| {
+    let (mms, data) = input;
+
+    #[cfg(feature = "debug")]
+    println!("input: {mms:?}, {data:?}");
+
+    let mut fragmentizer = Fragmentizer::new(mms);
+
+    // Ensure `Fragmentizer` recreates input data (when smaller than `mms`).
+    let mut emitted_bytes = Vec::new();
+    let check_recreation = match mms {
```

Comment:
Thought: Do we want to have `MaxMessageSize`? `Option<u32>` would be the same. Maybe even `u32` (with the meaning 0 is "unlimited") would be enough.

What I like about `MaxMessageSize` is that we can add good documentation. Maybe that is enough of a good reason.

Reply:
You can decide. I added it only for better readability.

```rust
let fragmentizer = Fragmentizer::new(MaxMessageSize::Unlimited);
```
looks better (and scarier) than
```rust
let fragmentizer = Fragmentizer::new(None);
```

Of course we also could add an `impl From<Option<u32>> MaxMessageSize`.

Reply:
Yeah, also thought about that. Another option would be to have a `Fragmentizer::new(u32)` and `Fragmentizer::unlimited()`. The limit could than be queried via `Fragmentizer::limit -> Option<u32>`.

Reply:
Can discuss later again.

Reply:
Please create an issue for that.

Reply:
Already did, see #541. (Forgot to mention in my last comment, sorry.)